### DISCUSSION TITLE
V18.0.0.0 upgrade gone wrong - exit-code

### DISCUSSION THREAD

**Discussion by blastbeatbilly** [awawa-dev/HyperHDR#383] [General]
Went to upgrade to V18 from V17 and I'm running into an issue where it's giving me an exit-code when I check the status of the service. I'm running a RPi 4 debian lite as running HyperHDR and Pi-Hole simultaneously. Tried going back to V17 with same issue listed. PLEASE tell me what I'm doing wrong here, I'm not exactly a savvy Linux user..

`pi@HyperHole:~ $ sudo service hyperhdr@pi status`
`● hyperhdr@pi.service - HyperHdr ambient light systemd service  for user pi`
`Loaded: loaded (/etc/systemd/system/hyperhdr@.service; enabled; vendor preset: enabled)`
`Active: activating (auto-restart) (Result: exit-code) since Wed 2022-10-19 19:37:07 EDT; 30ms ago`
`Process: 2130 ExecStart=/usr/bin/hyperhdr (code=exited, status=1/FAILURE)`
`Main PID: 2130 (code=exited, status=1/FAILURE)`

Edit: I used the 18.0.0.0 arm7l.deb file when I installed, followed the installation instructions to the letter.

**Comment by awawa-dev** [comment]
https://github.com/awawa-dev/HyperHDR/wiki/Installation#Linux

```
if the WWW panel is out of reach then run the following command to check what's happening:
$/usr/bin/hyperhdr -d
```

**Comment by blastbeatbilly** [comment]
This is the response I get:

```
pi@HyperHole:~ $ /usr/bin/hyperhdr -d
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /usr/bin/hyperhdr)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/hyperhdr/bin/../lib/libQt5Widgets.so.5)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.28' not found (required by /usr/share/hyperhdr/bin/../lib/libQt5Widgets.so.5)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /usr/share/hyperhdr/bin/../lib/libQt5Widgets.so.5)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/hyperhdr/bin/../lib/libQt5Gui.so.5)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/hyperhdr/bin/../lib/libQt5Core.so.5)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.28' not found (required by /usr/share/hyperhdr/bin/../lib/libQt5Core.so.5)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /usr/share/hyperhdr/bin/../lib/libQt5Core.so.5)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/hyperhdr/bin/../lib/libasound.so.2)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/hyperhdr/bin/../lib/libpng16.so.16)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/hyperhdr/bin/../lib/libicui18n.so.67)
/usr/bin/hyperhdr: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /usr/share/hyperhdr/bin/../lib/libicuuc.so.67)
```

**Comment by awawa-dev** [comment]
Your OS is probably too old. Check similar report from here: https://github.com/awawa-dev/HyperHDR/issues/338 And type ```ldd -version``` to get your glibc version. Nevertheless HyperHDR v17 from the official release should rather work (unless your Debian is really old ;) or you used improper installer for your architecture like armv7 for aarch64). Other solution than downgrading to v17 if you are not able to upgrade to Debian Bullseye, is to build v18 installer for your OS yourself (manual is in the wiki)
